### PR TITLE
WIP: fix setting velocity on paradropping cargo

### LINKED ISSUE #8199: ParadropItem inconsistency between SP and MP
**Mods (complete and add to the following information):**
- **Arma 3:** 2.02.147359
- **CBA:** 3.15.3
- **ACE3:** 3.13.6

**Description:**
Crates airdropped in SP are properly ([as scripted](https://github.com/acemod/ACE3/blob/76676eee462cb0bbe400a482561c148d8652b550/addons/cargo/functions/fnc_paradropItem.sqf#L53)) ejected from vehicle while in MP airdropping cargo results in crate spawning at correct position but with no velocity. Also parachute deploys before the crate can descend at all. It causes significantly different aiming requirements between SP and MP to hit the same spot.

**Steps to reproduce:**
1. For ease of testing set paradropTimeCoefficient to something low like 0.1 or 0.0.
2. Create a mission with GhostHawk and a crate that can be loaded into its cargo preferably in VR.
3. Run the mission in SP.
4. Load the crate into the helicopter.
5. Takeoff and hover 50m+
6. Airdrop the crate
7. Observe the crate go on a ballistic trajectory behind the helicopter until it starts drifting when parachute deploys.
8. Repeat steps 4.-6. in MP on dedicated server
9. Observe crate falling straight down before parachute opens

**Expected behavior:**
I would expect the crate to fall the same way in SP and MP.
**Where did the issue occur?**
-Dedicated vs Single Player
**Log Files:**
Not relevant

**Additional context:**
I observed this when working on airdropping tables.
**Screenshots:**
[Video](https://youtu.be/coppZnKVWsM) test on our (PGM) server with additional mods,  identical results were obtained on a locally hosted dedicated server with ACE and CBA only.

### PR DESCRIPTION
**Pull request by LupusTheCanine**
**When merged this pull request will:**
- Cargo - Fix setting velocity for airdropped objects

Fixes #8199

**Code changes summary:**
- Cargo - Add serverParadrop to handle invoking ownerParadrop
- Cargo - Add ownerParadrop to set velocity on owner's machine
- Cargo - Change velocity setting related code in paradropCargo.
- Cargo - replaced serverUnload event with serverParadrop

### CODE REVIEW

File: addons/cargo/XEH_postInit.sqf [hunk 1]
```diff
@@ -54,6 +54,21 @@
     [_item, "blockDamage", "ACE_cargo", false] call EFUNC(common,statusEffect_set);
 }] call CBA_fnc_addEventHandler;
 
+[QGVAR(ownerParadrop), {
+    params ["_item","_dropVelocity"];
+    
+    _item setVelocity _dropVelocity;
+}] call CBA_fnc_addEventHandler;
```
Comment (commy2) [inline addons/cargo/XEH_postInit.sqf:61]: This is just `QEGVAR(common,setVelocity)`.

Comment (LupusTheCanine) [reply addons/cargo/XEH_postInit.sqf]: done

File: addons/cargo/XEH_postInit.sqf [hunk 2]
```diff
@@ -54,6 +54,15 @@
     [_item, "blockDamage", "ACE_cargo", false] call EFUNC(common,statusEffect_set);
 }] call CBA_fnc_addEventHandler;
 
+[QGVAR(serverParadrop), {
+    params ["_item", "_emptyPosAGL","_dropVelocity"];
+
+    _item hideObjectGlobal false;
+    _item setPosASL (AGLtoASL _emptyPosAGL);
+    [QEGVAR(common,setVelocity), [_item,_dropVelocity],owner _item] call CBA_fnc_ownerEvent;
```
Comment (commy2) [inline addons/cargo/XEH_postInit.sqf:62]: ?

**Comment by commy2** [comment]
I do not understand how this is supposed to solve anything.

To me it seems like it should just be:

```sqf
_itemObject setVelocity ((velocity _vehicle) vectorAdd ((vectorNormalized (vectorDir _vehicle)) vectorMultiply -5));
```

->
```
private _velocity = velocity _vehicle vectorAdd (vectorNormalized vectorDir _vehicle vectorMultiply -5);
[QEGVAR(common,setVelocity), [_itemObject, _velocity]] call CBA_fnc_globalEvent;
```

and also consider the other `setVelocity` a few lines down.

Maybe I am missing something, but `ownerEvent` is a huge red flag to me.

**Comment by LupusTheCanine** [comment]
> I do not understand how this is supposed to solve anything.
> 
> To me it seems like it should just be:
> 
> ```sqf
> _itemObject setVelocity ((velocity _vehicle) vectorAdd ((vectorNormalized (vectorDir _vehicle)) vectorMultiply -5));
> ```
> 
> ->
> 
> ```sqf
> private _velocity = velocity _vehicle vectorAdd (vectorNormalized vectorDir _vehicle vectorMultiply -5);
> [QEGVAR(common,setVelocity), [_itemObject, _velocity]] call CBA_fnc_globalEvent;
> ```
> 
> and also consider the other `setVelocity` a few lines down.
> 
> Maybe I am missing something, but `ownerEvent` is a huge red flag to me.

After finalizing #8203 I will try a simpler approach you suggested in [#8199](https://github.com/acemod/ACE3/issues/8199#issuecomment-816636636).
What is the benefit of using global event vs target event? To me using global event looks like firing a claymore at a single soldier.

**Comment by commy2** [comment]
After rewatching the video, targetEvent seems to be sufficient, however, the current ownerEvent is just weird and there is no reason to make a second server handler on the server for a single line.

**Comment by LupusTheCanine** [comment]
Closing now, I have a new version which incorporates changes suggested by commy2 and implementing them in a new branch will yield much cleaner pull request.
I will create new pull request after finalizing #8203 as new fix is written with it in mind.

**Pull request closed without merging**